src/comps/orth.cc

```diff
@@ -0,0 +1,115 @@
+/*
+TODO #1: only store the upper triangle of the gram matrix in gram_vec,
+so that it can be of size k*(k+1)/2 instead of k*k.
+*/
+
+#include <lapack.hh>
+#include <RandBLAS.hh>
+#include <RandLAPACK.hh>
+
+namespace RandLAPACK::comps::orth {
+
+// Perfoms a Cholesky QR factorization
+template <typename T> 
+int Orth<T>::CholQR(
+        int64_t m,
+        int64_t k,
+        std::vector<T>& Q // pointer to the beginning
```

Comment:
I'm going to stop pointing this out, but ``// pointer to the beginning'' comments aren't needed in general and should be used sparingly.